contributing.rst

```diff
@@ -77,3 +77,63 @@ Developer Guidelines
     in C4.1 (e.g., imperative phrasing, wrap lines at 72 characters).
 
 For more details about C4.1, including commit requirements, see `RFC 1 <https://github.com/flux-framework/rfc/blob/master/spec_1.rst>`_.
+
+-------
+Testing
+-------
+
+Testing is an important part of development throughout all of the Flux
+components, and any changes or additions contributed should come with adequate
+testing. The below examples show how to run tests in a Flux project within
+their respective ``t/`` directories.
```

Comment:
Would it be worth mentioning that unit tests are generally co-located with sources and use `libtap`, and system tests are generally located under ``t/`` and use `sharness`?

Reply:
Also there are the python tests that use `pycotap`.

Might be helpful to add references to the third party test projects mentioned (libtap, pycotap, sharness)